### PR TITLE
ci: add k8s uninstallation to metrics tests

### PR DESCRIPTION
**Pull request by dborquez**
IPerf bench runs on top of a K8s setup, for instance ci metrics
needs to clean the k8s setup, before start the test.

Fixes: #4205

Signed-off-by: David Esparza <david.esparza.borquez@intel.com>

### CODE REVIEW

**Comment by GabyCT** [comment]
@dborquez I think this should be part of the commit where you are enabling iperf metrics because I have a lot of questions, 1. How frequent is the test failing in removing or installing k8s on the metrics CI? 2. How stable the results are? 3. Do we need k8s or ctr?

File: .ci/lib.sh
Comment (jodh-intel) [inline .ci/lib.sh:365]: Do we need a `sudo systemctl daemon-reload` here? It's possible (probable) that the `autoremove` below does that though?

File: .ci/lib.sh
Comment (jodh-intel) [inline .ci/lib.sh:362]: What about the other architectures? Would it be better to say `$(uname -m) != "x86_64"`?

**Comment by jodh-intel** [comment]
@dborquez - Please can you address the outstanding comments on this PR when you're back.

**Comment by jodh-intel** [comment]
Ping @dborquez.

**Comment by jodh-intel** [comment]
@dborquez - please can you update?

**Comment by dborquez** [comment]
@jodh-intel yes WIP

File: .ci/lib.sh
Comment (dborquez) [reply .ci/lib.sh:362]: thanks James

File: .ci/lib.sh
Comment (dborquez) [reply .ci/lib.sh:364]: @jodh-intel  thanks for the feedback, good point I pushed a fix

File: .ci/lib.sh
Comment (dborquez) [reply .ci/lib.sh]: @jodh-intel  thanks for the feedback, good point, so I pushed a fix

**Review by jodh-intel — APPROVED** [review]
Thanks @dborquez.

lgtm

**Comment by dborquez** [comment]
> @dborquez I think this should be part of the commit where you are enabling iperf metrics because I have a lot of questions, 1. How frequent is the test failing in removing or installing k8s on the metrics CI? 2. How stable the results are? 3. Do we need k8s or ctr?

@GabyCT thanks for the remarks.
The idea is to enable the iperf K8s version that already exist in metrics. See the issue #3475.
So this PR is to make the Metrics CI-job can remove K8s before to start the iperf test.
In a following PR I will enable iperf and then determine if the test can run continuously with no issues

**Comment by fidencio** [comment]
A head's up that may affect your PR.

In the Architecture Committee meeting from January 25th, 2022, the Architecture Committee has agreed on using the "Dismiss stale pull request approvals when new commits are pushed" configuration from GitHub.  It basically means that if your PR has been rebased or updated, the approvals given will be erased.

In order to minimize traumas due to the new approach, please, consider adding a note on the changes done before the rebase / force-push, and also pinging the reviewers for a subsequent round of reviews.

Thanks for your understanding!

Related issue: https://github.com/kata-containers/community/issues/249

**Comment by GabyCT** [comment]
> > @dborquez I think this should be part of the commit where you are enabling iperf metrics because I have a lot of questions, 1. How frequent is the test failing in removing or installing k8s on the metrics CI? 2. How stable the results are? 3. Do we need k8s or ctr?
> 
> @GabyCT thanks for the remarks. The idea is to enable the iperf K8s version that already exist in metrics. See the issue #3475. So this PR is to make the Metrics CI-job can remove K8s before to start the iperf test. In a following PR I will enable iperf and then determine if the test can run continuously with no issues

@dborquez currently the METRICS CI is installing and cleaning k8s without these additional steps that you are implementing here at this PR, we do not know if the cleaning and installing is being done without an issues because we do not have any test that we can prove that works fine, you can do another commit here in this PR where you enable the iperf test and then you will see if the clean that you are implementing here in this PR will actually work or otherwise we can merge this PR and later you will find out that is not working properly

**Comment by dborquez** [comment]
> 

@fidencio thank you, I'll follow the new approach.

**Review by GabyCT — APPROVED** [review]

**Comment by dborquez** [comment]
> > > @dborquez I think this should be part of the commit where you are enabling iperf metrics because I have a lot of questions, 1. How frequent is the test failing in removing or installing k8s on the metrics CI? 2. How stable the results are? 3. Do we need k8s or ctr?
> > 
> > 
> > @GabyCT thanks for the remarks. The idea is to enable the iperf K8s version that already exist in metrics. See the issue #3475. So this PR is to make the Metrics CI-job can remove K8s before to start the iperf test. In a following PR I will enable iperf and then determine if the test can run continuously with no issues
> 
> @dborquez currently the METRICS CI is installing and cleaning k8s without these additional steps that you are implementing here at this PR, we do not know if the cleaning and installing is being done without an issues because we do not have any test that we can prove that works fine, you can do another commit here in this PR where you enable the iperf test and then you will see if the clean that you are implementing here in this PR will actually work or otherwise we can merge this PR and later you will find out that is not working properly or it is working...but it is your call

@GabyCT thanks, I will add a new commit to this PR with the iperf enablement.

**Comment by dborquez** [comment]
> @dborquez if you want to test and run only metrics try with `/test-ubuntu-metrics` also if you comment the other tests that will fail because it will look for the results here https://github.com/kata-containers/tests/blob/main/cmd/checkmetrics/ci_worker/checkmetrics-json-qemu-kata-metric5.toml you need to also add there something for `iperf` like [0b3e935](https://github.com/kata-containers/tests/commit/0b3e93542d02947ac4dab15f44362a9c0e99d03c)

@ Thanks for the inputs. Improving the patch with recommended ci settings.

**Comment by GabyCT** [comment]
@dborquez the reason why metrics is failing because you need to add also a baseline for cloud hypervisor [https://github.com/kata-containers/tests/blob/main/cmd/checkmetrics/ci_worker](https://github.com/kata-containers/tests/blob/main/cmd/checkmetrics/ci_worker)

**Comment by dborquez** [comment]
> @dborquez the reason why metrics is failing because you need to add also a baseline for cloud hypervisor https://github.com/kata-containers/tests/blob/main/cmd/checkmetrics/ci_worker

@GabyCT yes I have added the baseline for iperf, thanks

File: .ci/lib.sh
Comment (GabyCT) [inline .ci/lib.sh:374]: disable that part

File: .ci/lib.sh
Comment (GabyCT) [inline .ci/lib.sh:377]: disable that part

**Review by GabyCT — CHANGES_REQUESTED** [review]

File: .ci/lib.sh
Comment (dborquez) [reply .ci/lib.sh:374]: lets see if that line fixes the issue. thanks

**Comment by jodh-intel** [comment]
Any update on this @dborquez ?

**Comment by dborquez** [comment]
> Any update on this @dborquez ?

WIP, debugging k8s that hangs up for ch metrics:
`error execution phase preflight: [preflight] Some fatal errors occurred:
01:34:45 	[ERROR Port-10250]: Port 10250 is in use`

File: .ci/lib.sh
Comment (jodh-intel) [inline .ci/lib.sh:368]: Can you remove this commented out code?

File: .ci/lib.sh
Comment (jodh-intel) [inline .ci/lib.sh:361]: Nit: The indentation in this block is too deep (one too many tabs fwics).

**Review by jodh-intel — APPROVED** [review]
Thanks @dborquez.

File: .ci/lib.sh
Comment (dborquez) [reply .ci/lib.sh]: @jodh-intel thanks, done

File: .ci/lib.sh
Comment (dborquez) [reply .ci/lib.sh]: @jodh-intel thank you, done

File: .ci/run.sh
Comment (GabyCT) [inline .ci/run.sh:105]: why do you remove this?

Comment (GabyCT) [reply .ci/run.sh:105]: I think is useful to have it to mark the cloud hypervisor start

Comment (dborquez) [reply .ci/run.sh:105]: Gaby, that info msg is located before the test starts, in the same way the msg for qemu does: https://github.com/kata-containers/tests/blob/f46c370026590300ad569f714728883a18a06933/.ci/run.sh#L114

**Pull request merged**
